saver.py

```diff
@@ -181,5 +189,30 @@ def process_step(self, step):
         if step % self.config['save_steps'] == 0:
             self.save_model(f'step{step}')
 
+        pending_save_best_loss = os.path.exists(os.path.join(self.save_root, ".pending_save_best_loss"))
+        if pending_save_best_loss:
+            self.save_model('best_loss')
+            if is_main_process():
+                if self.old_best is not None:
+                    print(f'New best loss: {self.best_loss} from {self.old_best} (Δ{self.old_best - self.best_loss} = {100 * (self.old_best - self.best_loss) / self.old_best:.2f}%)')
+                else:
+                    print(f'New best loss: {self.best_loss}')
+                os.replace(os.path.join(self.save_root, 'best_loss'), os.path.join(self.save_root, '.pending_save_best_loss'))
```

Comment:
I think this line got messed up, it should be os.replace(os.path.join(self.save_root, '.pending_save_best_loss'), os.path.join(self.save_root, 'best_loss.txt')) right?

Reply:
Crap. Yeah. Sorry, sloppy.